internal/db/update/schema.go

```diff
@@ -6,20 +6,31 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
-	"sort"
 	"strings"
 
 	"github.com/canonical/lxd/lxd/db/query"
 	"github.com/canonical/lxd/lxd/db/schema"
 	"github.com/canonical/lxd/shared"
 )
 
+// updateType represents whether the update is an internal or external schema update.
+type updateType uint
+
+const (
+	// updateInternal represents an internal schema update from microcluster.
+	updateInternal updateType = 0
+
+	// updateExternal represents a schema update external to microcluster, applied after all internal updates.
+	updateExternal updateType = 1
+)
+
+// SchemaUpdate holds the configuration for executing schema updates.
 type SchemaUpdate struct {
-	updates []schema.Update // Ordered series of updates making up the schema
-	hook    schema.Hook     // Optional hook to execute whenever a update gets applied
-	fresh   string          // Optional SQL statement used to create schema from scratch
-	check   schema.Check    // Optional callback invoked before doing any update
-	path    string          // Optional path to a file containing extra queries to run
+	updates map[updateType][]schema.Update // Ordered series of internal updates making up the schema
```

Comment:
Internal updates? It presumably also supports external ones too due to the updateType key?